### PR TITLE
Manage queue unit tests

### PR DESCRIPTION
**Pull request by dbox-fnal**
unit tests for manage_queue.py, Trace.py, and mpq.py

To Be Done:
1) The changes from lists to sets in recent commit 1f1ebf70c1821de2862e3150ce28c1f7629a4c4b broke several unit tests, which were written prior to the commit. I am working on fixing the unit tests
2) Atomic_Request_Queue::get_sg() is revealed as broken as currently implemented, a fix is suggested in the unit test comments
3) I am investigating Atomic_Request_Queue::update() as it seems to do unnecessary deletes followed by re-inserting identical requests in its constituent queues

### CODE REVIEW

**Review by renbauer-fnal — APPROVED** [review]

File: src/tests/test_manage_queue.py [hunk 1]
```diff
@@ -157,124 +159,126 @@ def test_rm(self):
...
         q2 = self.q.get_queue(queue_key='opt')
         self.assertTrue(isinstance(q2, list))
-        #test a bad key
+        # test a bad key
         with mock.patch('sys.stderr', new=StringIO.StringIO()) as _out:
             try:
                 q3 = self.q.get(queue_key='bad_key')
                 self.assertTrue(False)
-            except TypeError, KeyError:
+            except TypeError as KeyError:
```
Comment (renbauer-fnal) [inline src/tests/test_manage_queue.py:271]: I don't think this is valid, is it?

Comment (dbox-fnal) [reply src/tests/test_manage_queue.py:271]: Its valid, but could be clearer.
Perhaps something like
except TypeError, KeyError as an_exception:
     #an_exception is of an expected type, TypeError or KeyError
except:
     # shouldnt get here, complain and fail test

**Comment by ddbox** [comment]
Merged.  The To Be Done issues will follow shortly

**Comment by ddbox** [comment]
not sure why the merge failed, trying again

**Pull request merged**